player/configfiles.c

```diff
@@ -313,7 +320,7 @@ void mp_write_watch_later_conf(struct MPContext *mpctx)
         goto exit;
     }
 
-    mp_mk_config_dir(mpctx->global, MP_WATCH_LATER_CONF);
+    mp_mk_config_dir(mpctx->global, mpctx->cached_watch_later_configdir);
```

Comment:
From a superficial look at the code, it looks like calling mp_mk_config_dir after mp_get_playback_resume_config_filename in mp_write_watch_later_conf might be fine. But it's up to you to verify this.